CorvallisBus.Core/Models/DaysOfWeek.cs

```diff
@@ -79,17 +79,31 @@ public static DaysOfWeek GetDaysOfWeek(string days)
         }
 
         /// <summary>
-        /// Returns a value indicating whether the provided DaysOfWeek value is applicable today.
+        /// Returns a value indicating whether the provided DateTimeOffset falls in to the DaysOfWeek specified
         /// </summary>
-        public static bool IsToday(DaysOfWeek days, TimeSpan lastArrivalTime, DateTimeOffset currentTime)
+        public static bool TodayMayFallInsideDaySchedule(BusStopRouteDaySchedule ds, DateTimeOffset currentTime)
         {
-            var normalizedArrivalTime = lastArrivalTime.Days == 1 ? lastArrivalTime - TimeSpan.FromDays(1) : lastArrivalTime;
+            DaysOfWeek currentDay = ToDaysOfWeek(currentTime.DayOfWeek);
 
-            // if the last arrival time is later than the current time, this pushes the "effective day" back by one.
-            // For example Sunday at 1 AM will have an "effective day" of Saturday when looking for the right Night Owl schedule.
-            var effectiveDay = currentTime - normalizedArrivalTime;
+            // simple case where current time is definitely inside the current schedule
+            return (ds.Days & currentDay) == currentDay || TimeInSpilloverWindow(ds, currentTime);
+        }
+
+        /// <summary>
+        /// Returns true if the current time is in the early morning of e.g. Tuesday, and this day schedule contains 'late night runs' for a Monday schedule that spill over in to Tuesday morning
+        /// </summary>
+        public static bool TimeInSpilloverWindow(BusStopRouteDaySchedule ds, DateTimeOffset currentTime)
+        {
+            DaysOfWeek previousDay = ToDaysOfWeek(currentTime.AddDays(-1).DayOfWeek);
+            TimeSpan lastTime = ds.Times.Last();
+
+            if ((ds.Days & previousDay) == previousDay && lastTime.Days >= 1)
+            {
+                DateTimeOffset lastScheduleDateTime = new DateTimeOffset(currentTime.Year, currentTime.Month, currentTime.Day, lastTime.Hours, lastTime.Minutes, 0, 0, currentTime.Offset);
```

Comment:
this doesn't need to change but it is too bad if this is the best way to get "this date, and this time of day"